File: templates/express-prisma/src/Routes/UserRoute.js
import { Router } from "express";
import {LoginUser,RegisterUser,LogoutUser} from "../Controllers/UserController.js"
import AuthUser from "../Middlewares/AuthMiddelware.js"
const UserRouter=Router()

UserRouter.get("/",(req,res)=>{
    return res.send("users endpoint is up")
})

UserRouter.post("/register",RegisterUser)
UserRouter.post("/login",LoginUser)
UserRouter.get("/logout",AuthUser,LogoutUser)


export default UserRouter

File: templates/go-http/internal/routes/userRoute.go
package routes

import (
	"net/http"

	controller "http-server/internal/controllers"
	"http-server/internal/middlewares"
)

func UserRouter(app *http.ServeMux, ctrl *controller.Controller) {
	app.HandleFunc("POST /users/register", ctrl.RegisterHandler)
	app.HandleFunc("POST /users/login", ctrl.LoginHandler)
	app.HandleFunc("GET /users/me", middleware.Auth(ctrl.MeHandler))
	app.HandleFunc("POST /users/logout", ctrl.LogoutHandler)
}
